File: app/src/main/java/com/student/rentals/ui/fragments/home_fragment/HomeViewModel.java
package com.student.rentals.ui.fragments.home_fragment;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.student.Api.NetworkModule;
import com.student.models.DataApartmentList;

import javax.inject.Inject;

import retrofit2.Call;
import retrofit2.Callback;
import timber.log.Timber;

public class HomeViewModel extends ViewModel {
    @Inject public HomeViewModel() { }
    @Inject NetworkModule networkModule;

    private MutableLiveData<DataApartmentList> dataList;

    public LiveData<DataApartmentList> getRoomsList() {
        if (dataList == null) {
            dataList = new MutableLiveData<DataApartmentList>();
            loadDatas();
        }
        return dataList;
    }


    private void loadDatas() {

        Call<DataApartmentList> call = new NetworkModule().getApiClient().getAllApartments();
        call.enqueue(new Callback<DataApartmentList>() {
            @Override
            public void onResponse(Call<DataApartmentList> call, retrofit2.Response<DataApartmentList> response) {
                dataList.setValue(response.body());
            }

            @Override
            public void onFailure(Call<DataApartmentList> call, Throwable t) {

            }
        });

    }
}

File: app/src/main/java/com/student/rentals/ui/fragments/home_fragment/HomeViewModel.kt
package com.student.rentals.ui.fragments.home_fragment

import android.service.autofill.UserData
import android.util.Log
import androidx.lifecycle.*
import com.student.Api.ApiInterface
import kotlinx.coroutines.launch

class HomeViewModel(private val apiInterface: ApiInterface) : ViewModel() {
    val TAG:String = "HomeViewModel"
    private val _uiState = MutableLiveData<UserData>()
    val uiState: LiveData<UserData> =_uiState
    init {
        retrieveMovies()
    }


    fun retrieveMovies(){
        viewModelScope.launch {
            kotlin.runCatching {
                apiInterface.getUser()

            }.onSuccess {
                Log.d(TAG,"Hurressssssssssssssssss")
            }.onFailure {
                Log.d(TAG, "Faileediiiiiiiiii")
            }
        }
    }
}
